### ISSUE TITLE
provide validation for destination

### ISSUE THREAD

**Issue by linsun** [istio/istio#12794]
https://discuss.istio.io/t/istio-1-1-ingressgateway-not-working-on-a-fresh-install/1581/5

I'm surprised our validation didn't fail for user's VS yaml.

**Comment by esnible** [comment]
As pointed out in the Discuss thread, the VirtualService has

```
      route:
      - destination:
          host: default/productpage
```

The Host is unusual because it contains a namespace.  Our documentation at https://istio.io/docs/reference/config/networking/v1alpha3/virtual-service/#Destination says the `host` should be a Kubernetes or Consul service name.


This produces an Envoy Route:
```
          "route": {
           "cluster": "outbound|80||default/productpage.default.svc.cluster.local",
```

but the Envoy Cluster has a different name:

```
     "cluster": {
      "name": "outbound_.9080_._.productpage.default.svc.cluster.local",
```

The bug is that we call ValidateWildcardDomain() for the host, but don't actually tolerate wildcards.

**Comment by esnible** [comment]
@frankbu Before you close this, the new logic allows

```
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  name: productpage
spec:
  hosts:
  - productpage
  http:
  - route:
    - destination:
        host: *
        subset: v1
```

That destination seems invalid.

**Comment by frankbu** [comment]
@esnible The new logic is unchanged, it's just fixing a bug where a number of invalid cases where not failing the validation.

The wildcard syntax is allowed in the destination host to handle the case where the destination is an external service defined in a ServiceEntry. So whatever `host` is allowed in a ServiceEntry is allowed in a route destination. I know we've gone back and forth as to whether you can define a ServiceEntry for host `*`, but unless we decide it is definitely not supported, we should leave the validator as it is.

@rshriram do you think we should tighten up the validator to not allow `*` for a route destination?

**Comment by frankbu** [comment]
Fixed in #13222
